functions/src/notifications/deliverNotifications.ts

```diff
@@ -106,6 +106,95 @@ const deliverEmailNotifications = async () => {
   // Wait for all email documents to be created
   await Promise.all(emailPromises)
 }
+type ProfileData = {
+  email: string
+  fullName: string
+  role: string
+}
+
+export const deliverOrgUpgradeStatus = functions.firestore
+  .document("profiles/{userId}")
+  .onWrite(async (snapshot, context) => {
+    const before = snapshot.before.data() as ProfileData
+    const after = snapshot.after.data() as ProfileData
+
+    if (before.role === "rolePending") {
```

Comment:
If the profile hasn't been created yet, `before` will be undefined and the function will throw an error.